core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/impl/InstallPackageServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+package com.exadel.etoolbox.backpack.core.services.pckg.impl;
+
+import com.exadel.etoolbox.backpack.core.dto.response.PackageInfo;
+import com.exadel.etoolbox.backpack.core.dto.response.PackageStatus;
+import com.exadel.etoolbox.backpack.core.services.LoggerService;
+import com.exadel.etoolbox.backpack.core.services.SessionService;
+import com.exadel.etoolbox.backpack.core.services.pckg.BasePackageService;
+import com.exadel.etoolbox.backpack.core.services.pckg.InstallPackageService;
+import com.exadel.etoolbox.backpack.core.services.pckg.PackageInfoService;
+import com.exadel.etoolbox.backpack.core.servlets.model.InstallPackageModel;
+import org.apache.jackrabbit.vault.fs.api.ProgressTrackerListener;
+import org.apache.jackrabbit.vault.fs.io.ImportOptions;
+import org.apache.jackrabbit.vault.packaging.DependencyHandling;
+import org.apache.jackrabbit.vault.packaging.JcrPackage;
+import org.apache.jackrabbit.vault.packaging.JcrPackageManager;
+import org.apache.jackrabbit.vault.packaging.PackageException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.Calendar;
+
+/**
+ * Implements {@link InstallPackageService} to provide install package operations
+ */
+@Component(service = InstallPackageService.class)
+public class InstallPackageServiceImpl implements InstallPackageService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InstallPackageServiceImpl.class);
+    private static final String SERVICE_NAME = "backpack-service";
+    private static final String START_INSTALL_MESSAGE = "Install Package: ";
+
+    @Reference
+    private PackageInfoService packageInfoService;
+
+    @Reference
+    private BasePackageService basePackageService;
+
+    @Reference
+    private SessionService sessionService;
+
+    @Reference
+    private LoggerService loggerService;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PackageInfo installPackage(ResourceResolver resourceResolver, InstallPackageModel installPackageModel) {
+        PackageInfo packageInfo = packageInfoService.getPackageInfo(resourceResolver, installPackageModel);
+        if (!PackageStatus.INSTALL_IN_PROGRESS.equals(packageInfo.getPackageStatus()) && !PackageStatus.BUILD_IN_PROGRESS.equals(packageInfo.getPackageStatus())) {
+            packageInfo.setPackageStatus(PackageStatus.INSTALL_IN_PROGRESS);
+            packageInfo.clearLog();
+            packageInfo.addLogMessage(START_INSTALL_MESSAGE + packageInfo.getPackagePath());
+            packageInfo.addLogMessage(LocalDateTime.now().toString());
```

Comment:
Is it right to add just a date on a new line?

Reply:
thanks, I left the time start as in the default package manager